File: app/src/main/java/com/jx/androiddemo/contract/main/MainContract.java
package com.jx.androiddemo.contract.main;

import com.jx.androiddemo.BaseView;
import com.jx.androiddemo.bean.main.MainPageBean;
import com.jx.androiddemo.presenter.BasePresenter;

import java.util.List;

public class MainContract {
    public interface View extends BaseView {

    }

    public interface Presenter extends BasePresenter<View> {
        void test();
    }
}

File: app/src/main/java/com/jx/androiddemo/contract/main/MainContract.kt
package com.jx.androiddemo.contract.main

import com.jx.androiddemo.BaseView
import com.jx.androiddemo.presenter.BasePresenter

class MainContract {
    interface View : BaseView {
        fun showMsg(str: String)
    }

    interface Presenter : BasePresenter {
        fun test()
        fun test2()
    }
}